test/integration/uds_integration_test.h

```diff
@@ -11,14 +13,25 @@
 #include "gtest/gtest.h"
 
 namespace Envoy {
-class UdsIntegrationTest : public HttpIntegrationTest,
-                           public testing::TestWithParam<Network::Address::IpVersion> {
+class UdsIntegrationTest
+    : public HttpIntegrationTest,
+      public testing::TestWithParam<std::tuple<Network::Address::IpVersion, bool>> {
 public:
-  UdsIntegrationTest() : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+  UdsIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, std::get<0>(GetParam())),
+        abstract_namespace_(std::get<1>(GetParam())) {
+#if !defined(__linux__)
+    if (abstract_namespace_) {
+      EXPECT_THROW("Abstract AF_UNIX sockets are only supported on linux.", EnvoyException);
+    }
+#endif
+  }
 
   void createUpstreams() override {
     fake_upstreams_.emplace_back(new FakeUpstream(
-        TestEnvironment::unixDomainSocketPath("udstest.1.sock"), FakeHttpConnection::Type::HTTP1));
+        abstract_namespace_ ? "@/my/udstest"
```

Comment:
nit: add a function to retrieve the uds path for the current test config, and use it in the two places this snippet is copied.